### ISSUE TITLE
[Enhancement]: Make it clear what you can edit on the checkout blocks

### ISSUE THREAD

**Issue by passatgt** [woothemes/woocommerce#49395]
### Describe the solution you'd like

I receive multiple support tickets asking how they can rename these options on the checkout page:

![image (7)](https://github.com/woocommerce/woocommerce/assets/43415/db6eed02-1f28-4c78-8a8c-0fc9eb5b3818)

You can just simply edit these when you edit the block, but for some reason it's not obvious to many users that you can basically change every text on this page. Some UI clue would be nice i think, for example showing a pencil icon next to the labels, or underline the text with a dotted line? At least until the first time they edit it. And i think many don't realize you can actually change the step names/description too. 

<img width="793" alt="image" src="https://github.com/woocommerce/woocommerce/assets/43415/eeb9ade4-0cf6-4d01-b0b8-e736a28bc7e0">

**Comment by opr** [comment]
Thanks a lot @passatgt. I agree some clue that these are editable would be super helpful.

I've added this to our backlog! 

@nikkivias do you have any recommendations on how we can signal to the merchant these are editable strings?

**Comment by pierorocca** [comment]
@yjailin @elizaan36 flagging this community feedback for the design team. It's a good idea for consideration at the editor level.

**Comment by elizaan36** [comment]
An editing affordance would be helpful since all text can be edited. I think this is something that would have to happen at the editor level as you mentioned so merchants can learn only one affordance for all editable text, so we could create a proposal but it would ideally be aligned with work upstream.

**Comment by pierorocca** [comment]
> An editing affordance would be helpful since all text can be edited. I think this is something that would have to happen at the editor level as you mentioned so merchants can learn only one affordance for all editable text, so we could create a proposal but it would ideally be aligned with work upstream.

Makes sense thanks @elizaan36. Could you take that forward as VoC upstream for consideration or point me to who I can share this feedback with?

**Comment by elizaan36** [comment]
This is interesting because there is an affordance on other areas of editable text in Gutenberg. The blue outline appears as an indication that the content inside can be editable. I recorded a [video](https://www.loom.com/share/919a822dd8464107b2f60ecbf1695a70?sid=ace5ca51-f88e-4505-b21c-da275a49e0ec) to illustrate.

@opr I see you commented on here way back when. Have we discussed using the outline affordance on editable areas of the checkout block like the Delivery toggle labels? I don't recall if there's a reason not to, and I could see it being really helpful to solve this confusion. There's no reason to diverge from how others areas of the site are presented.

**Comment by opr** [comment]
> Have we discussed using the outline affordance on editable areas of the checkout block like the Delivery toggle labels? I don't recall if there's a reason not to, and I could see it being really helpful to solve this confusion. There's no reason to diverge from how others areas of the site are presented.

Don't think we discussed it, but yes it makes a lot of sense and I don't see a reason not to. Thanks Elizabeth

**Comment by pierorocca** [comment]
@yjailin this hasn't been looked at in a while. I think the guidance still stands that Gutenberg already solved for this and I see that desired behaviour in the template editor. In the checkout page editor however, the user has to randomly click to discover the editable blocks. Could you weigh in?

**Comment by yjailin** [comment]
@pierorocca  @elizaan36 I investigated why the blue outline doesn't appear on "Ship" and "Pickup" in the checkout editor, using Claude. The labels are already RichText fields (technically editable) but they're not registered as individual blocks. Gutenberg's blue outline only appears on block boundaries, so since these labels live inside custom button markup with no block boundary around them, there's nothing for the editor to outline.

Three options to fix it:
A. Make each label a real inner block. Full native affordance, but structural change with BC risk.
B. Add a CSS hover/focus outline to the labels. Lightweight, solves discoverability without restructuring, no new pattern needed.
C.  Surface the fields in the sidebar inspector. No inline affordance but merchants can find them in settings.

Option A is def not the way to go, we already have enough nested blocks.
Option C is like introducing a new editing pattern.
Option B seems to be best call and doesn't required any new/custom pattern.

**Comment by elizaan36** [comment]
Option B makes sense if we can align with the Gutenberg editing affordance. It might drift over time if there are changes but I think it's worth the fix.

**Comment by senadir** [comment]
I think pattern B make sense, it's not a new thing that a block has editable content
